### ISSUE TITLE
crash when scrolling fast with prefetch enabled

### ISSUE THREAD

**Issue by felixjichao** [tilltue/TLPhotoPicker#77]
the index paths returned by prefetchItemsAt seem to be unordered, so **first...last** could not form a valid Range.

```
open func collectionView(_ collectionView: UICollectionView, prefetchItemsAt indexPaths: [IndexPath]) {
        if self.usedPrefetch {
            queue.async { [weak self] in
                guard let `self` = self, let collection = self.focusedCollection else { return }
                if indexPaths.count <= collection.count,let first = indexPaths.first?.row, let last = indexPaths.last?.row {
                    guard let assets = collection.getAssets(at: first...last) else { return }
                    let scale = max(UIScreen.main.scale,2)
                    let targetSize = CGSize(width: self.thumbnailSize.width*scale, height: self.thumbnailSize.height*scale)
                    self.photoLibrary.imageManager.startCachingImages(for: assets, targetSize: targetSize, contentMode: .aspectFill, options: nil)
                }
            }
        }
    }
```

**Comment by tilltue** [comment]
Fixed 08bfbcfbaf02fe335e8b4a13c5ace31ba8887cf5

**Comment by felixjichao** [comment]
there is a similar issue within 
`func collectionView(_ collectionView: UICollectionView, cancelPrefetchingForItemsAt indexPaths: [IndexPath])`

**Comment by tilltue** [comment]
Fixed 08bfbcfbaf02fe335e8b4a13c5ace31ba8887cf5
